### PR TITLE
Auth: OAuth token sync improvements

### PR DESCRIPTION
**Pull request by mgyongyosi**
**What is this feature?**
This PR improves the way the OAuth access token is refreshed using the refresh token by creating a separate context derived from context.Background.

Besides, it introduces a new metric called `token_refresh_fetch_duration_seconds` to track the duration of the refresh tokens.

**Why do we need this feature?**
To improve the observability of this functionality.

**Who is this feature for?**

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/oauthtoken/oauth_token.go [hunk 1]
```diff
@@ -17,6 +17,7 @@ import (
 	"github.com/grafana/grafana/pkg/services/login"
 	"github.com/grafana/grafana/pkg/services/user"
 	"github.com/grafana/grafana/pkg/setting"
+	"github.com/prometheus/client_golang/prometheus"
```
Comment (kalleep) [inline pkg/services/oauthtoken/oauth_token.go:20]: Nit: Fix import order

File: pkg/services/oauthtoken/oauth_token.go [hunk 2]
```diff
@@ -212,8 +216,12 @@ func (o *Service) tryGetOrRefreshAccessToken(ctx context.Context, usr *login.Use
 
 	persistedToken := buildOAuthTokenFromAuthInfo(usr)
 
+	start := time.Now()
 	// TokenSource handles refreshing the token if it has expired
 	token, err := connect.TokenSource(ctx, persistedToken).Token()
+	duration := time.Since(start)
+	o.tokenRefreshDuration.WithLabelValues(authProvider, fmt.Sprintf("%t", err == nil)).Observe(duration.Seconds())
```
Comment (kalleep) [inline pkg/services/oauthtoken/oauth_token.go:223]: Do we only want to capture the refresh duration for the metric or would it make more sense to capture the duration of the whole function?

**Review by kalleep — APPROVED** [review]
LGTM, left some nits that is not blocking

File: pkg/services/oauthtoken/oauth_token.go (hunk 2 above)
Comment (mgyongyosi) [reply pkg/services/oauthtoken/oauth_token.go:223]: Yes, with this I'd to capture the roundtrip time of the outgoing request.

File: pkg/services/oauthtoken/oauth_token.go (hunk 1 above)
Comment (mgyongyosi) [reply pkg/services/oauthtoken/oauth_token.go]: Fixed, thanks!

**Pull request merged**